src/NServiceBus.Core/Pipeline/satellitePipelineExecutor.cs

```diff
@@ -17,7 +14,12 @@ public SatellitePipelineExecutor(IBuilder builder, SatelliteDefinition definitio
 
         public Task Invoke(MessageContext messageContext)
         {
+            messageContext.Context.Set(messageContext.TransportTransaction);
```

Comment:
We would have to change the persister seam to remove this need => we decided to do that in vNext